src/Orleans.Core/Messaging/IGatewayListProvider.cs

```diff
@@ -14,8 +12,7 @@ public interface IGatewayListProvider
         /// <summary>
         /// Initializes the provider, will be called before all other methods
         /// </summary>
-        /// <param name="clientConfiguration">the given client configuration</param>
-        Task InitializeGatewayListProvider(ClientConfiguration clientConfiguration);
+        Task InitializeGatewayListProvider();
```

Comment:
I think we still need this async init method, since User may need to do some async action to initialize their gatewayListProvider. 
Although there's discussion on change from async init method to factory pattern. But I think that is a major pattern change which is out of scope of this PR

Reply:
Ok